backend/src/routes/user.rs

```diff
@@ -286,3 +317,261 @@ pub async fn user_log_in(
+        .one(&pg_con)
+        .await
+    {
+        Ok(opt_state) => match opt_state {
+            Some(state) => {
+                let valid_burrows = match get_burrow_list(state.valid_burrow.clone()).await {
+                    Ok(burrows_id) => burrows_id,
+                    Err(e) => {
+                        error!(
+                            "[GET BURROW] Failed to get valid burrows: {:?}",
+                            e.to_string()
+                        );
+                        return (Status::InternalServerError, Json(Vec::new()));
+                    }
+                };
+                let banned_burrows = match get_burrow_list(state.banned_burrow.clone()).await {
+                    Ok(burrows_id) => burrows_id,
+                    Err(e) => {
+                        error!(
+                            "[GET BURROW] Failed to get valid burrows: {:?}",
+                            e.to_string()
+                        );
+                        return (Status::InternalServerError, Json(Vec::new()));
+                    }
+                };
+                let burrows_id = [valid_burrows, banned_burrows].concat();
+                let mut response = Vec::new();
+                for burrow_id in burrows_id {
+                    match pgdb::burrow::Entity::find_by_id(burrow_id)
+                        .one(&pg_con)
+                        .await
+                    {
+                        Ok(opt_burrow) => match opt_burrow {
+                            Some(burrow) => {
+                                response.push(UserGetBurrowResponse {
+                                    id: burrow.burrow_id,
+                                    title: burrow.title.clone(),
+                                    description: burrow.description.clone(),
+                                    post_num: burrow.post_num,
+                                });
+                            }
+                            None => {
+                                error!("[GET-BURROW] Cannot find burrow by burrow_id.");
+                                return (Status::InternalServerError, Json(Vec::new()));
+                            }
+                        },
+                        Err(e) => {
+                            error!("[GET-BURROW] Database Error: {:?}", e.to_string());
+                            return (Status::InternalServerError, Json(Vec::new()));
+                        }
+                    };
+                }
+                (Status::Ok, Json(response))
+            }
+            None => {
+                error!("[GET BURROW] Cannot find user_status by uid.");
+                return (Status::InternalServerError, Json(Vec::new()));
+            }
+        },
+        Err(e) => {
+            error!("[GET BURROW] Database Error: {:?}", e.to_string());
+            return (Status::InternalServerError, Json(Vec::new()));
+        }
+    }
+}
+
+#[get("/favorite")]
+pub async fn get_favorite(
```

Comment:
Better change this to collection.

We will return what user collection has, but don't return what user likes.

And I don't think we need to return burrow?

Reply:
Here I returned as the doc says. Might be changed later

Reply:
copy to have a collection interface either maybe.